model/config.go

```diff
@@ -2064,19 +2062,8 @@ func (mes *MessageExportSettings) isValid(fs FileSettings) *AppError {
 			return NewAppError("Config.IsValid", "model.config.is_valid.message_export.daily_runtime.app_error", nil, "", http.StatusBadRequest)
 		} else if _, err := time.Parse("15:04", *mes.DailyRunTime); err != nil {
 			return NewAppError("Config.IsValid", "model.config.is_valid.message_export.daily_runtime.app_error", nil, err.Error(), http.StatusBadRequest)
-		} else if mes.FileLocation == nil {
-			return NewAppError("Config.IsValid", "model.config.is_valid.message_export.file_location.app_error", nil, "", http.StatusBadRequest)
 		} else if mes.BatchSize == nil || *mes.BatchSize < 0 {
 			return NewAppError("Config.IsValid", "model.config.is_valid.message_export.batch_size.app_error", nil, "", http.StatusBadRequest)
-		} else if *fs.DriverName != IMAGE_DRIVER_LOCAL {
-			if absFileDir, err := filepath.Abs(fs.Directory); err != nil {
-				return NewAppError("Config.IsValid", "model.config.is_valid.message_export.file_location.relative", nil, err.Error(), http.StatusBadRequest)
-			} else if absMessageExportDir, err := filepath.Abs(*mes.FileLocation); err != nil {
-				return NewAppError("Config.IsValid", "model.config.is_valid.message_export.file_location.relative", nil, err.Error(), http.StatusBadRequest)
-			} else if !strings.HasPrefix(absMessageExportDir, absFileDir) {
-				// configured export directory must be relative to data directory
-				return NewAppError("Config.IsValid", "model.config.is_valid.message_export.file_location.relative", nil, "", http.StatusBadRequest)
-			}
```

Comment:
This was the real reason for this change. See [this thread](https://pre-release.mattermost.com/core/pl/i1sowdf44pdd8kz7a8kf4gnbmo) for the reasoning behind the change.